tools/cli/internal/openapi/oasdiff.go

```diff
@@ -160,6 +164,69 @@ func removeExternalRefs(path *openapi3.PathItem) *openapi3.PathItem {
 	return path
 }
 
+// handlePathConflict handles the path conflict by checking if the conflict should be skipped or not.
+func (o OasDiff) handlePathConflict(basePath *openapi3.PathItem, basePathName string) error {
+	if !o.shouldSkipPathConflict(basePath, basePathName) {
+		return errors.PathConflictError{
+			Entry: basePathName,
+		}
+	}
+
+	var pathsAreIdentical bool
+	var err error
+	if pathsAreIdentical, err = o.arePathsIdenticalWithExcludeExtensions(basePathName); err != nil {
+		return err
+	}
+
+	log.Printf("Skipping conflict for path: %s, pathsAreIdentical: %v", basePathName, pathsAreIdentical)
+	if pathsAreIdentical {
+		return nil
+	}
+
+	// allowDocsDiff = true not supported
+	if allOperationsAllowDocsDiff(basePath) {
+		return errors.AllowDocsDiffNotSupportedError{
+			Entry: basePathName,
+		}
+	}
+
+	d, err := o.noExtDiff.GetPathDiffWithoutExtensions(o.base.Spec, o.external.Spec)
+	if err != nil {
+		return err
+	}
+
+	return errors.PathDocsDiffConflictError{
+		Entry: basePathName,
+		Diff:  d,
+	}
+}
+
+// shouldSkipConflict checks if the conflict should be skipped.
+// The method goes through each path operation and performs the following checks:
+// 1. Validates if both paths have same operations, if not, then it returns false.
+// 2. If both paths have the same operations, then it checks if there is an x-xgen-soa-migration annotation.
```

Comment:
Shouldn't the logic check for x-gen-soa-migration even if the operations are not identical? Scenario:
MMS has a resource with 4 operations and the standalone service has a resource with only one operation which has `x-xgen-soa-migration`. This scenario should be quite common as teams might migrate one operation at a time instead of all operations. Is this something that we should support?  
Let me know I am missing something and this scenario is covered

Reply:
Yes! :) You're right, I completely forgot that for GetOasPaths we give them the method + path, so it's posisble a path exists in both services. I'll just double-check with SRE, since our initial premisse was "no duplicated paths" (so maybe we'd ask teams to move everything all at once)

Reply:
It looks like if the answer is yes for my question, we'll also need to consider the path conflicts without x-xgen-soa-migration right? so I'll open a ticket for this functionality

Reply:
[CLOUDP-277871](https://jira.mongodb.org/browse/CLOUDP-277871)